src/item_location.cpp

```diff
@@ -102,14 +126,14 @@ class item_location::item_on_map : public item_location::impl
         }
 
         void remove_item() override {
-            if( what == nullptr ) {
+            if( !target() ) {
```

Comment:
It might be best if the error reporting is added to `target` as that will trigger for any use of the class